### ISSUE TITLE
Posts pagination numbers are just numbers

### ISSUE THREAD

**Issue by afercia** [WordPress/twentytwentyone#475]
**Describe the bug**
The Posts pagination numbers are just numbers, thus they're announced by screen readers just as, for example:

"2 link", "3 link", etc.

![Screenshot (153)](https://user-images.githubusercontent.com/1682452/95979995-ce6c2880-0e1c-11eb-9fff-60c01d9c6ed7.png)

Other bundled themes, for example Twenty Seventeen, add some visually hidden text to prepend the word "Page" and make the link text more meaningful for screen reader users:

![Screenshot (154)](https://user-images.githubusercontent.com/1682452/95980014-d4620980-0e1c-11eb-9fb7-c9c35f16cf62.png)

However, this would impact sighted speech recognition users. They would see a link with a visible text "2" and they would try to voice a command like "Click 2". The command would fail because the actual accessible name of the link would be "Page 2".

The only option to cover all needs would be to make the "Page" text visible, for example:

![page](https://user-images.githubusercontent.com/1682452/95987854-4855df00-0e28-11eb-9f2a-67733fc96cd5.png)

I do realize this would impact design, adding the Design label.

**Comment by enriquesanchez** [comment]
We could place these links inside a `role=group` so they are announced with context to screen readers:

"link 2, Posts pagination, group"

<img width="772" alt="Screen Shot 2020-10-14 at 12 00 04" src="https://user-images.githubusercontent.com/3276087/96021515-2acb4a00-0e15-11eb-806b-6278d8fa0cbc.png">

Another similar idea is to have these links `aria-describedby` by a heading, label or hidden label.

If we do this both the design and users of voice recognition should not be affected.

**Comment by ryelle** [comment]
The links are already in a `nav role="navigation"` with label "Posts", but that label isn't announced with each link. Would adding a group to just the numbers work better? There is also already a hidden heading ("Posts navigation"), we could attach that using `aria-describedby`, but would that be picked up by the link list menu?

**Comment by enriquesanchez** [comment]
@ryelle Ah, you're right. `role="navigation"` makes more sense here, and this means the link is announced as:

"Link 2, Posts, navigation"

I personally think this is enough context.

>but that label isn't announced with each link

Right, VoiceOver only announces the `role` on the first item you focus on in the list. As far as I can tell, this is standard behavior and avoids unnecessary verbosity and repetition.

I think we could still change the label to "Posts pagination" so it's a bit clearer what the navigation is about. What do you think?

**Comment by carolinan** [comment]
Should the default wording be changed in core?
https://developer.wordpress.org/reference/functions/get_the_posts_pagination/

**Comment by afercia** [comment]
Just to note that playing with a role=group or aria-describedby won't help when these links are read out of context. Screen readers provide a few ways to get a list of the links in a page or jump through links etc. and in these cases the links are "out of context".

For example, NVDA has a tool called "Elements list" which, amongst other things, lists all the links in a page:

![Screenshot (157)](https://user-images.githubusercontent.com/1682452/96222968-74d73b80-0f8d-11eb-9d40-f63400ed45ab.png)

This feature is very similar to the VoiceOver "rotor" that can be invoked by pressing Control Option U: once there, use the left-right arrow keys to select the panel you're interested in. One of them lists all the links.

As said, these are examples. There are more scenarios where links can be read "out of context". In these cases the only way to make a link meaningful is by using a meaningful text.

Also, I'd like to remind that ARIA should not be used when there are other native ways to solve an issue. See the first rule of ARIA use: https://www.w3.org/TR/using-aria/#rule1

**Comment by carolinan** [comment]
<del>There is no parameter for adding text to the numbers in core, do we also want to open a trac issue?</del>
I did not look far enough, there is paginate_links: https://developer.wordpress.org/reference/functions/paginate_links/

**Comment by afercia** [comment]
There is 🙂 At least for `the_posts_pagination()` and it's used in Twenty Seventeen:  `before_page_number`
